gpiozero/input_devices.py

```diff
@@ -515,6 +525,15 @@ def __init__(
             self.close()
             raise
 
+    @property
+    def value(self):
+        """
+        Returns a value representing the average of the queued values. This
+        is nearer 0 for black under the sensor, and nearer 1 for white under
+        the sensor.
```

Comment:
Apologies for being a bit blunt (I'm tired), but "nearer 0" and "nearer 1" sounds a bit vague and wishy-washy?
I'm probably missing something obvious, but surely the sensor is either detecting a line, or not detecting a line? It's not probabilistic, is it? :confused: 

Or if it _is_ probabilistic, maybe this would be clearer if phrased more explicitly like "A value less than 0.5 means there's probably black under the sensor, and a value above 0.5 means there's probably white under the sensor"?
IMHO "a value less than 0.5" seems easier to understand than "a value nearer 0" :man_shrugging: 

BTW does black being ~0 and white being ~1 mean you changed your mind again about the "logic" behind `.value`? (that you were discussing in another issue)

I googled and found https://www.instructables.com/id/Using-IR-Sensor-TCRT-5000-With-Arduino-and-Program/ which suggests turning on and off the emitter, to help remove ambient noise from the receiver? I guess that would be hardware-incompatible with this current LineSensor class, but maybe it'd be worth adding a DenoisedLineSensor in future which did that? :smiley:

Reply:
Yeah, it is wishy-washy and the logic is indeed (still) backwards. Haven't changed my mind: it's wrong - but it's a non-backwards-compatible change (if anyone's relying on the current "polarity" of value) so we need to be a bit careful changing it. The nearer 0/1 thing is because it's a "smoothed" input device. The sensor does indeed just return 0/1 but we queue up those values as fast as we can and return an average of them. Historically this was a mean (hence the nearer) - now I think it's a actually a median so with a pure digital input it *should* also be a digital output (assuming an odd-length queue).

Anyway, for the next point release, this is something to come back to and clean up properly :)